.github/workflows/build-and-test.yml

```
@@ -0,0 +1,75 @@
+name: build-and-test
+on:
+  push:
+    branches: [main]
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+  pull_request:
+
+env:
+  GO_VERSION: 1.23.8
```

nit: why not using go 1.24.2? 

```suggestion
  GO_VERSION: 1.24.2
```

---

I had originally wanted to stay consistent with this repo's `GO_VERSION` usage, and the [workflow](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/.github/workflows/functional_test_v2.yaml) I'd been referencing was using `1.23.6`. Now that you point this out, I see the version isn't consistent across this repo's workflows, so I'll update. Thanks!